cumulus/taskflow/core/__init__.py

```diff
@@ -0,0 +1,239 @@
+         'config': {
+            'launch': {
+                'spec': launch_spec,
+                'params': launch_params
+            },
+            'provision': {
+                'spec': provision_spec
+            }
+        },
+        'profileId': cluster['profileId']
+    }
+    client = create_girder_client(
+        task.taskflow.girder_api_url, task.taskflow.girder_token)
+
+    try:
+        cluster = client.post('clusters',  data=json.dumps(body))
+    except HttpError as he:
+        task.logger.exception(he.responseText)
+        raise
+
+    msg = 'Created cluster: %s' % cluster['_id']
+    task.taskflow.logger.info(msg)
+    task.logger.info(msg)
+
+    # Now save cluster id in metadata
+    task.taskflow.set_metadata('cluster', cluster)
+
+    task.logger.info('Starting cluster.')
+
+    body = {
+        'status': 'launching'
+    }
+    client.patch('clusters/%s' % cluster['_id'], data=json.dumps(body))
+
+    secret_key = profile['secretAccessKey']
+    profile = profile.copy()
+    del profile['secretAccessKey']
+    log_write_url = '%s/clusters/%s/log' % (task.taskflow.girder_api_url,
+                                            cluster['_id'])
+    provision_params['cluster_state'] = 'running'
+    launch_params['cluster_state'] = 'running'
+    girder_token = task.taskflow.girder_token
+    cumulus.ansible.tasks.cluster.start_cluster(
+        launch_spec, provision_spec, cluster, profile, secret_key,
+        launch_params, provision_params, girder_token, log_write_url)
+
+    # Get the update to date cluster
+    cluster = client.get('clusters/%s' % cluster['_id'])
+
+    return cluster
+
+def create_girder_client(girder_api_url, girder_token):
+    client = GirderClient(apiUrl=girder_api_url)
+    client.token = girder_token
+
+    return client
+
+def _get_image(logger, profile, image_spec):
+    # Fetch the image from the CloudProvider
+    provider = CloudProvider(profile)
+    images = provider.get_machine_images(owner=image_spec['owner'],
+                                         tags=image_spec['tags'])
+
+    if len(images) == 0:
+        raise Exception('Unable to locate machine image for the ' +
+                        'following spec: %s' % image_spec)
+    elif len(images) > 1:
+        logger.warn('Found more than one machine image for the ' +
+                    'following spec: %s' % image_spec)
+
+    return images[0]['image_id']
+
+def terminate_jobs(task, client, cluster, jobs):
+    for job in jobs:
+        task.logger.info('Terminating job %s' % job['_id'])
+        # Fetch the latest job info
+        job_url = 'jobs/%s' % job['_id']
+        job = client.get(job_url)
+
+        # Update the status to terminating
+        body = {
+            'status': JobState.TERMINATING
+        }
+        client.patch(job_url, data=json.dumps(body))
+
+        terminate_job(
+            cluster, job, log_write_url=None,
+            girder_token=task.taskflow.girder_token)
```

Comment:
Now that we have this code in cumulus. I don't think it belongs in taskflow/core/__init__.py Lets move it to taskflow/clusters/ or something like that?